Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/self_intersections.h

```diff
@@ -503,11 +504,21 @@ bool does_self_intersect(const FaceRange& face_range,
     CGAL::Emptyset_iterator unused_out;
     internal::self_intersections_impl<ConcurrencyTag>(face_range, tmesh, unused_out, true /*throw*/, np);
   }
-  catch(CGAL::internal::Throw_at_output_exception&)
+  catch (const CGAL::internal::Throw_at_output_exception&)
   {
     return true;
   }
-
+  #if defined(CGAL_LINKED_WITH_TBB) && defined(TBB_USE_CAPTURED_EXCEPTION)
```

Comment:
No, that is wrong. Here is the correct expression:
```suggestion
  #if defined(CGAL_LINKED_WITH_TBB) && TBB_USE_CAPTURED_EXCEPTION
```
In my version of TBB (version 2020.2), `tbb_config.h` defines `TBB_USE_CAPTURED_EXCEPTION` as `0` (the false value) or `1` (the true value). Either way it is defined, and `defined(TBB_USE_CAPTURED_EXCEPTION)` will always evaluate to `1`.

Reply:
Damn. Ok, this is what is not working on my platform and if I use `!TBB_USE_CAPTURED_EXCEPTION` it works on macOS but gives warnings on yours.

Reply:
I tried to define other TBB macros and no luck either.

Reply:
I can try to use `!TBB_USE_CAPTURED_EXCEPTION` and `__clang__` to get rid of the warning maybe on your platform?

Reply:
It seems brew packages TBB version 2020.3. And the `tbb_config.h` of that version has the same logic as on my machine:
https://github.com/oneapi-src/oneTBB/blob/eca91f16d7490a8abfdee652dadf457ec820cc37/include/tbb/tbb_config.h#L488-L500.

What does not work with `#if defined(CGAL_LINKED_WITH_TBB) && TBB_USE_CAPTURED_EXCEPTION`?

Reply:
```
TBB Warning: Exact exception propagation is requested by application but the linked library is built without support for it
libc++abi.dylib: terminating with uncaught exception of type tbb::captured_exception: std::exception
Abort trap: 6
```

Reply:
Because brew version of TBB seems to have been compiled with a compiler that is not the default compiler of the machine, you should add `-DTBB_USE_CAPTURED_EXCEPTION=1` to the `CMAKE_CXX_FLAGS` of that test platform, to fix that issue.

Reply:
Yes, this does work! Thanks a lot!